test/james.data.constraints.tests.pas

```diff
@@ -109,11 +109,11 @@ procedure TDataStreamTest.AsString;
   Buf := TMemoryStream.Create;
   Ss := TStringList.Create;
   try
-    Buf.WriteBuffer(TXT[1], Length(TXT) * SizeOf(Char));
+    Buf.WriteBuffer(AnsiString(TXT)[1], Length(TXT));
```

Comment:
I tested it before committing. The problem remains.
However, if s is declared as AnsiString, it does work.
If you'd rather haver the compiler do some implicit casting instead of explicit, like above, it can be done.